tapis-project/tapis-ui
# task/TUI-114 -- components for arrays of dict fields and file inputs

## Overview:

Implement basic fileInputs array support

## Related Github Issues:

- [TUI-114](https://github.com/tapis-project/tapis-ui/issues/114)

## Summary of Changes:

- FieldArray wrappers for useFieldArray logic
- Added Collapse container component
- Added FileInputs component

## Testing Steps:

1. Log in with cicsvc account
2. Navigate to SlurmSleepSecondsTaccProd-0.0.1 app
3. Expand and collapse FileInputs
4. Try changing field contents
5. Try adding file inputs (only non-required fields are removable, so any additional ones you add will be but the default 3 will not be.)
6. Hit submit, see console for data structure produced

## UI Photos:

![image](https://user-images.githubusercontent.com/17181582/133189791-4999c15d-348d-4d29-b130-d63f360984d5.png)


## Notes:

- meta object not yet implemented
